From 0934b8fc91e3f08369d98fb9dcc24a1a5521a132 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Wed, 4 Jun 2025 22:30:07 +0100
Subject: [PATCH 1/4] test_analyze_wheel_abi: refactor test with explicit env

Be explicit in the test-case matrix when to set env variable. This
enables to check if environment variable is actually correctly reset.

This commit passes unit tests.
---
 tests/integration/test_bundled_wheels.py | 25 ++++++++++++++----------
 1 file changed, 15 insertions(+), 10 deletions(-)

diff --git a/tests/integration/test_bundled_wheels.py b/tests/integration/test_bundled_wheels.py
index db440ff..28c7d2d 100644
--- a/tests/integration/test_bundled_wheels.py
+++ b/tests/integration/test_bundled_wheels.py
@@ -2,12 +2,10 @@ from __future__ import annotations
 
 import importlib
 import json
-import os
 import sys
 import zipfile
 from argparse import Namespace
 from datetime import datetime, timezone
-from os.path import isabs
 from pathlib import Path
 from typing import Any
 from unittest.mock import Mock
@@ -30,53 +28,60 @@ HERE = Path(__file__).parent.resolve()
             "cffi-1.5.0-cp27-none-linux_x86_64.whl",
             {"libffi.so.5", "libpython2.7.so.1.0"},
             frozenset(),
+            None,
         ),
         (
             "cffi-1.5.0-cp27-none-linux_x86_64.whl",
             set(),
             frozenset(["libffi.so.5", "libpython2.7.so.1.0"]),
+            None,
         ),
         (
             "cffi-1.5.0-cp27-none-linux_x86_64.whl",
             {"libffi.so.5", "libpython2.7.so.1.0"},
             frozenset(["libffi.so.noexist", "libnoexist.so.*"]),
+            None,
         ),
         (
             "cffi-1.5.0-cp27-none-linux_x86_64.whl",
             {"libpython2.7.so.1.0"},
             frozenset(["libffi.so.[4,5]"]),
+            None,
         ),
         (
             "cffi-1.5.0-cp27-none-linux_x86_64.whl",
             {"libffi.so.5", "libpython2.7.so.1.0"},
             frozenset(["libffi.so.[6,7]"]),
+            None,
         ),
         (
             "cffi-1.5.0-cp27-none-linux_x86_64.whl",
             {"libpython2.7.so.1.0"},
             frozenset([f"{HERE}/*"]),
+            "LD_LIBRARY_PATH",
         ),
         (
             "cffi-1.5.0-cp27-none-linux_x86_64.whl",
             {"libpython2.7.so.1.0"},
             frozenset(["libffi.so.*"]),
+            None,
         ),
-        ("cffi-1.5.0-cp27-none-linux_x86_64.whl", set(), frozenset(["*"])),
+        ("cffi-1.5.0-cp27-none-linux_x86_64.whl", set(), frozenset(["*"]), None),
         (
             "python_snappy-0.5.2-pp260-pypy_41-linux_x86_64.whl",
             {"libsnappy.so.1"},
             frozenset(),
+            None,
         ),
     ],
 )
-def test_analyze_wheel_abi(file, external_libs, exclude):
+def test_analyze_wheel_abi(file, external_libs, exclude, env):
     # If exclude libs contain path, LD_LIBRARY_PATH need to be modified to find the libs
     # `lddtree.load_ld_paths` needs to be reloaded for it's `lru_cache`-ed.
-    modify_ld_library_path = any(isabs(e) for e in exclude)
 
     with pytest.MonkeyPatch.context() as cp:
-        if modify_ld_library_path:
-            cp.setenv("LD_LIBRARY_PATH", f"{HERE}")
+        if env:
+            cp.setenv(env, f"{HERE}")
             importlib.reload(lddtree)
 
         winfo = analyze_wheel_abi(
@@ -88,10 +93,10 @@ def test_analyze_wheel_abi(file, external_libs, exclude):
             allow_graft=True,
         )
         assert set(winfo.external_refs["manylinux_2_5_x86_64"].libs) == external_libs, (
-            f"{HERE}, {exclude}, {os.environ}"
+            f"{HERE}, {exclude}, {env}"
         )
 
-    if modify_ld_library_path:
+    if env:
         importlib.reload(lddtree)
